src/lambdas/controller/random-get.ts

```diff
@@ -0,0 +1,63 @@
+import fetch, { Response } from 'node-fetch'
+import { flatten } from 'lodash'
+import { Controller } from './baseController'
+import { RandomBreedsRequest, AllBreeds, ExtendReponse } from '../types'
+import { ExternalRequest } from '../types/requestType'
+
+import { FetchBreedError } from '../errors'
+
+class RandomGetController extends Controller implements ExternalRequest {
+  randomBreedRequest: RandomBreedsRequest
+
+  async makeRequest(url: string) {
+    const res: Response = await fetch(url)
+    this.setStatus(res.status)
+    if (res.status >= 200 && res.status <= 299) {
+      return res
+    }
+    throw new FetchBreedError()
+  }
+
+  async handleRequest(response: Response) {
+    this.randomBreedRequest = await response.json()
+  }
+
+  flattenAllBreeds() {
+    const allBreeds: AllBreeds = flatten(
+      Object.entries(this.randomBreedRequest.message).map(([key, value]) => {
+        if (value.length) {
+          return Object.values(value).map((v) => {
+            return `${v} ${key}`
+          })
+        }
+        return key
+      }),
+    )
+    return allBreeds
+  }
+}
+
+export async function handler(): Promise<ExtendReponse<AllBreeds>> {
+  try {
+    const requestBreeds = new RandomGetController()
+    const response = await requestBreeds.makeRequest('https://dog.ceo/api/breeds/list/all')
+    await requestBreeds.handleRequest(response)
+    const payload = await requestBreeds.flattenAllBreeds()
+    return {
+      statusCode: requestBreeds.getStatus(),
+      body: payload,
+    }
+  } catch (err: unknown) {
+    if (err instanceof FetchBreedError) {
+      return {
+        statusCode: 500,
+        errorCode: err.errorCode,
+        message: err.message,
+      }
```

Comment:
Might be better to mask the raw error messages here as we are unsure of who might be consuming the results of this handler. 

So for example, instead of leaking the message through with `err.message` we could have a static expressive string for each error case.

Reply:
hmm so I've thrown a customer error for FetchBreedError with an internal error code. Do you mean doing a switch case by each status code >= 400 ?

Reply:
Sorry, I missed that you had the custom error with message, that should work great.